packages/client/hmi-client/src/services/charts.ts

```diff
@@ -804,3 +805,179 @@ export function createInterventionChart(
 	}
 	return spec;
 }
+
+/// /////////////////////////////////////////////////////////////////////////////
+// Funman charts
+/// /////////////////////////////////////////////////////////////////////////////
+
+enum BoxType {
+	Satisfactory = 'Satisfactory',
+	Unsatisfactory = 'Unsatisfactory',
+	Ambiguous = 'Ambiguous',
+	ModelChecks = 'Model checks'
+}
+
+export function createFunmanStateChart(data: ProcessedFunmanResult, constraints: any[], stateId: string) {
+	const globalFont = 'Figtree';
+
+	const boxLines = data.trajs.map((traj) => {
+		const boxType = traj.label === 'true' ? BoxType.Satisfactory : BoxType.Unsatisfactory;
+		return { timepoints: traj.timestep, value: traj[stateId], boxType };
+	});
+
+	// Find min/max values to set an appropriate viewing range for y-axis
+	const minY = Math.floor(Math.min(...boxLines.map((d) => d.value)));
+	const maxY = Math.ceil(Math.max(...boxLines.map((d) => d.value)));
+
+	// Show checks for the selected state
+	const stateIdConstraints = constraints.filter((c) => c.variables.includes(stateId));
+	const modelChecks = stateIdConstraints.map((c) => ({
+		boxType: BoxType.ModelChecks,
+		startX: c.timepoints.lb,
+		endX: c.timepoints.ub,
+		// If the interval bounds are within the min/max values of the line plot use them, otherwise use the min/max values
+		startY: Math.max(c.additive_bounds.lb, minY),
+		endY: Math.min(c.additive_bounds.ub, maxY)
+	}));
```

Comment:
Yeah I decided to just focus on the line rather than the model checks. I set it up like this because if we choose greater than the upper bound is super high and with less than the lower bound is super low. In those cases you can't have a good idea of the trajectory of line as it either goes in a straight horizontal line or in some cases the graph doesn't render properly (just empty space).